Automattic/woocommerce-payments
# Return `_payment_method_id` for subscription payment meta.

Fixes https://github.com/Automattic/woocommerce-subscriptions-core/issues/128.

#### Changes proposed in this Pull Request









#### Testing instructions





* Create a new subscription order.
* As a customer, add a new payment method.
* Make the new payment method the default.
* When prompted "Would you like to update your subscriptions to use this new payment method", select Yes to update subscriptions to the new payment method.
* Go to My Account > Subscriptions > See payment method for all the subscriptions are updated as expected.

note:
- I notice a bug (or maybe by design) and plan to create a separate issue for it that settings a default payment method only applies to subscriptions with payment token of the same gateway. I see that it's because gateway id is passed to `WCS_Payment_Tokens::get_customer_tokens()` [here](https://github.com/Automattic/woocommerce-subscriptions-core/blob/trunk/includes/class-wcs-my-account-payment-methods.php#L172).
- Also, I notice a bug and plan to create a separate issue for it that when all payment methods are deleted, then adding a new payment won't update the existing subscriptions' payment token. I think it's because we look for subscriptions based on existing tokens [here](https://github.com/Automattic/woocommerce-subscriptions-core/blob/trunk/includes/class-wcs-my-account-payment-methods.php#L222-L232). When payments are deleted and new tokens are added, this logic fails because it assumes subscriptions' payment tokens are part of the existing tokens.
- I am also still working on adding e2e test for this PR change.

-------------------


**Post merge**
